### PR TITLE
util/types: fix parsing datatime 00-00-00

### PR DESCRIPTION
**Pull request by tiancaiamao**
00-00-00 should not be adjust year to 2000-00-00

It should be:

```
mysql> create table t2 (a int, t datetime);
Query OK, 0 rows affected (0.19 sec)

mysql> insert into t2 values (1, "00-00-00");
Query OK, 1 row affected (0.00 sec)

mysql> select * from t2;
+------+---------------------+
| a    | t                   |
+------+---------------------+
|    1 | 0000-00-00 00:00:00 |
+------+---------------------+
1 row in set (0.00 sec)
```

But we got:

```
mysql> select * from t2;
+------+---------------------+
| a    | t                   |
+------+---------------------+
|    1 | 2000-00-00 00:00:00 |
+------+---------------------+
1 row in set (0.01 sec)
```

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @XuHuaiyu @zimulala @shenli

File: util/types/time_test.go [hunk 1]
```diff
@@ -48,6 +48,8 @@ func (s *testTimeSuite) TestDateTime(c *C) {
 		{"20121231113045", "2012-12-31 11:30:45"},
 		{"121231113045", "2012-12-31 11:30:45"},
 		{"2012-02-29", "2012-02-29 00:00:00"},
+		{"00-00-00", "0000-00-00 00:00:00"},
+		{"00-00-00 00:00:00.123", "2000-00-00 00:00:00"},
```
Comment (XuHuaiyu) [inline util/types/time_test.go]: why not 2000-00-00 00:00:00.123? @tiancaiamao

**Comment by zimulala** [comment]
LGTM

File: util/types/time_test.go (hunk 1 above)
Comment (tiancaiamao) [inline util/types/time_test.go]: Because `ParseDataTime` use DefaultFsp which is 0, fsp part is ignored. @XuHuaiyu

**Pull request merged**